base_classes/NXcalibration.nxdl.xml

```diff
@@ -0,0 +1,220 @@
+        </doc>
+    </field>
+    <field name="applied" type="NX_BOOLEAN">
+        <doc>
+             Has the calibration been applied?
+        </doc>
+    </field>
+    <field name="calibration_software">
+        <doc>
+             Name of the software used for this calibration.
+        </doc>
+        <attribute name="version">
+            <doc>
+                 Software version.
+            </doc>
+        </attribute>
+    </field>
+    <field name="original_axis" type="NX_FLOAT" units="NX_ANY">
+        <doc>
+             Vector containing the data coordinates in the original uncalibrated axis
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="ncal"/>
+        </dimensions>
+        <attribute name="symbol">
+            <doc>
+                 The symbol of the axis to be used in the fit_function, e.g., `energy`, `E`.
+                 This should comply to the following naming rules (similar to python's naming rules):
+                 
+                   * A variable name must start with a letter or the underscore character
+                   * A variable name cannot start with a number
+                   * A variable name can only contain alpha-numeric characters and underscores (A-z, 0-9, and _ )
+                   * Variable names are case-sensitive (age, Age and AGE are three different variables)
+            </doc>
+        </attribute>
+        <attribute name="input_path">
+            <doc>
+                 The path from which this data is derived, e.g., raw detector axis.
+                 Should be a valid NeXus path name, e.g., /entry/instrument/detector/raw.
+            </doc>
+        </attribute>
+    </field>
+    <field name="input_SYMBOL" type="NX_FLOAT" units="NX_ANY">
+        <doc>
+             Additional input axis to be used in the formula.
+             The part after `input_` is used as the symbol to be used in the `fit_function`, i.e.,
+             if the field name is `input_my_field` you should refer to this axis by `my_field` in the `fit_function`.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="ncal"/>
+        </dimensions>
+        <attribute name="input_path">
+            <doc>
+                 The path from which this data is derived, e.g., raw detector axis.
+                 Should be a valid NeXus path name, e.g., /entry/instrument/detector/raw.
+            </doc>
+        </attribute>
+    </field>
+    <field name="coefficients" type="NX_FLOAT" units="NX_ANY">
+        <doc>
+             For non-linear energy calibrations, e.g. in a TOF, a polynomial function is fit
+             to a set of features (peaks) at well defined energy positions to determine
+             E(TOF). Here we can store the array of fit coefficients.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="ncoeff"/>
+        </dimensions>
+    </field>
+    <field name="fit_function" type="NX_CHAR">
```

Comment:
Following discussion in NXresolution, a similar compromise would be ok for me

Reply:
Probably also remove attribute `symbol` in `input_data`

Reply:
Use formula description instead of formula

Reply:
Using `NXparameters` and formula description from now on